project/src/main/java/com/google/sps/servlets/Announcements.java

```diff
@@ -0,0 +1,97 @@
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Date;
+ 
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/** Servlet that posts and loads announcemnts*/
+@WebServlet("/announcement")
+public class Announcements extends HttpServlet {
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  UserService userService = UserServiceFactory.getUserService();
+  GsonBuilder gsonBuilder = new GsonBuilder();
+  Gson gson = gsonBuilder.create();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String name = "";
+      String school = "";
+      Query query = new Query("User");
+      PreparedQuery results = datastore.prepare(query);
+      for(Entity entity:results.asIterable()){
+          if (((String) entity.getProperty("email")).toLowerCase().equals(userService.getCurrentUser().getEmail().toLowerCase())) {
+              name = (String) entity.getProperty("name");
+              school = (String) entity.getProperty("school");
+          }
+      }
+      String title = request.getParameter("title");
+      String content = request.getParameter("content");
+      Entity commentEntity = new Entity("Announcements");
+      java.util.Date now=new java.util.Date();
+      commentEntity.setProperty("from", name);
+      commentEntity.setProperty("title", title);
+      commentEntity.setProperty("content", content);
+      commentEntity.setProperty("when", now);
+      commentEntity.setProperty("school", school);
+      datastore.put(commentEntity);
+  }
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      ArrayList<Object> data = new ArrayList<Object>();
+      String school = "";
+      Query query = new Query("User");
+      PreparedQuery results = datastore.prepare(query);
+      for(Entity entity:results.asIterable()){
+          if (((String) entity.getProperty("email")).toLowerCase().equals(userService.getCurrentUser().getEmail().toLowerCase())) {
+              school = (String) entity.getProperty("school");
+          }
+      }
+      query = new Query("Announcements").addSort("when", SortDirection.DESCENDING);
+      results = datastore.prepare(query);
+      for(Entity entity:results.asIterable()){
+          if (((String) entity.getProperty("school")).toLowerCase().equals(school.toLowerCase())) {
+              ArrayList<Object> pair = new ArrayList<Object>();
+              pair.add((String) entity.getProperty("from"));
+              pair.add((String) entity.getProperty("title"));
+              pair.add((String) entity.getProperty("content"));
+              data.add(pair);
+          }
+      }
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(data));
+      //this will get all announcements entities, with school = current user school
```

Comment:
make the comment on the top?

Reply:
and where did you parse and use "current school"?

Reply:
For some reason my git fetch ended up pulling Fernando's previous branch. I did not implement the announcements.java file... I am not sure why it ended up in my pull request.

Reply:
There are three commits that somehow got intertwined with Fernando's pull request

Reply:
ok. Please make sure you only merge the last 3 commits.